src/main/java/com/feywild/feywild/block/model/MagicalBrazierModel.java

```diff
@@ -0,0 +1,29 @@
+package com.feywild.feywild.block.model;
+
+import com.feywild.feywild.FeywildMod;
+import com.feywild.feywild.block.MagicalBrazierBlock;
+import com.feywild.feywild.block.entity.MagicalBrazier;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.world.level.block.state.BlockState;
+import software.bernie.geckolib3.model.AnimatedGeoModel;
+
+public class MagicalBrazierModel extends AnimatedGeoModel<MagicalBrazier> {
+
+    @Override
+    public ResourceLocation getModelLocation(MagicalBrazier magicalBrazier) {
+        return new ResourceLocation(FeywildMod.getInstance().modid, "geo/magical_brazier.geo.json");
+    }
+
+    @Override
+    public ResourceLocation getTextureLocation(MagicalBrazier magicalBrazier) {
+        BlockState state = magicalBrazier.getBlockState();
+        if (MagicalBrazierBlock.isLit(state)) {
+            return new ResourceLocation(FeywildMod.getInstance().modid, "textures/block/magical_brazier" + magicalBrazier.getTextureNumber() + ".png");
```

Comment:
For performance reasons, please create a public static final list that holds all the resource locations and then access elements from that.

This can be done with
```java
public static final List<ResourceLocation> TEXTURE_IDS = IntStream.range(0, textureCount).mapToObject(i -> FeywildMod.getInstance().resource("textures/block/magical_brazier" + i + ".png")).toList();
```

Reply:
` public static List<ResourceLocation> TEXTURE_IDS = IntStream.range(1, 8).mapToObj(i -> FeywildMod.getInstance().resource("textures/block/magical_brazier" + i + ".png")).toList();
`

`
 @Override
    public ResourceLocation getTextureLocation(MagicalBrazier magicalBrazier) {
        BlockState state = magicalBrazier.getBlockState();
        if (MagicalBrazierBlock.isLit(state)) {
            return TEXTURE_IDS.get(magicalBrazier.getTextureNumber());
        } else {
            return new ResourceLocation(FeywildMod.getInstance().modid, "textures/block/magical_brazier.png");
        }
    }
`
` public int getTextureNumber() {

        return this.textureNumber;
    }

    @Override
    public void tick() {
        if (level != null && level.isClientSide) {
            if (!(this.animationCount == 2)) {
                this.animationCount++;
            } else {
                this.animationCount = 0;

                if (!(this.textureNumber == 8)) {
                    this.textureNumber++;
                } else {
                    this.textureNumber = 1;

                }
            }
        }
    }`

This gives me following error: 
`java.lang.ArrayIndexOutOfBoundsException: Index 8 out of bounds for length 8`

Reply:
As texture numbers start at 1, use `TEXTURE_IDS.get(magicalBrazier.getTextureNumber() - 1)` and use `IntStream.range(1, textureCount + 1)` when creating the list